feat(browser-starfish): fetch span domain in dropdown and allow filtering

DominikB2014 opened the pull request:
Changes
1. Span domain is now populating dropdown
2. We can filter by span domain
![image](https://github.com/getsentry/sentry/assets/44422760/484a7701-5576-4e01-8c90-31e33960f662)

3. some other cleanup items

AbhiPrasad commented on static/app/views/performance/browser/resources/index.tsx:
```diff
@@ -143,9 +151,10 @@ function PageSelector({value}: {value?: string}) {
-  });
+  const options: Option[] = [
+    {value: '', label: 'All'},
+    ...pages.map(page => ({value: page, label: page})),
```
Would be super cool to get the favicon of the site if possible (should be available under `DOMAIN_NAME/favicon.ico` - see https://avatars.githubusercontent.com/favicon.ico).

Might be jank in darkmode though 🤔

DominikB2014 replied:
Wow, that's an awesome idea!

DominikB2014 replied:
I'm gonna go through this and all these other PR idea and write them all down in one task!

AbhiPrasad commented on static/app/views/performance/browser/resources/utils/useResourceDomansQuery.ts:
```diff
@@ -0,0 +1,48 @@
+  const {slug: orgSlug} = useOrganization();
+  const {transaction} = useResourceModuleFilters();
+
+  const fields = ['avg(span.self_time)', 'span.domain']; // TODO: this query fails without avg(span.self_time)
```
m: can we use `SpanMetricsField.SPAN_DOMAIN`? Likewise with all other cases.

I think we need an aggregation like `avg`. We used `count()` for the domain selector on the database page. Not sure what is faster though 🤔.

DominikB2014 replied:
Intuitively count sounds like it would be faster, because the simplest calculation for average includes count() avg = sum(duration)/count(). But maybe clickhouse does something under the hood 🤷 
I'll switch this to count just to be consistent

AbhiPrasad commented on static/app/views/performance/browser/resources/utils/useResourceDomansQuery.ts:
```diff
@@ -0,0 +1,48 @@
+    limit: 100,
+  });
+  const data: string[] =
+    result?.data?.data.map(row => row['span.domain'][0].toString()).sort() || [];
```
l: can we capture a sentry error here if we see that length of `row['span.domain']` is `>1`? This should never happen for resource spans, but it'll help spot future regressions.

DominikB2014 replied:
Good idea!

AbhiPrasad reviewed (commented):
Let's try to use the `SpanMetricsField` enums wherever possible here, will ✅ after that change, the rest is up to you!

DominikB2014 commented:
@AbhiPrasad updated to use `SpanMetricsFields` everywhere, also found a bunch of places outside of this PR (within resource module), that needed the update, so added that as well.
Also addressed your comments, I like the ideas!

AbhiPrasad commented on static/app/views/performance/browser/resources/utils/useResourceDomansQuery.ts:
```diff
@@ -0,0 +1,61 @@
+      .map(row => {
+        const domains = row[SPAN_DOMAIN] as any as string[];
+        if (domains?.length > 1) {
+          Sentry.captureException(new Error('More than one domain found in a resource'));
```
can we add the domain to the error message?

DominikB2014 replied:
Yeah, good idea, will followup with that on Monday!
